### PR TITLE
Add common ruleset help files.

### PR DESCRIPTION
**Pull request by jwrober**
Document what the varying flags do for units, buildings and techs.

### CODE REVIEW

**Review by lmoureaux — CHANGES_REQUESTED** [review]
The PR itself looks interesting, but I want to understand the bigger picture before going down the road. How does this integrate in the ToC tree, how does it interact with Modding/Rulesets, how will we keep the rst files updated, when will we update them, where all this will be documented, how we ensure that `pyciv` remains stable enough, ...

**Comment by jwrober** [comment]
Right now these files do not integrate in the TOC. They are effectively hidden right now. `sphinx_fc2_manuals` references them. When the python script runs, it places all of the ruleset details to `docs/Manuals/Rulesets/{ruleset}` and then varying pages inside the ruleset docs will reference these common files as "help" links.

I also reference some files in the Modding section. However these pages are more about helping as a form of documentation and are not really in scope from a modding perspective. They are aimed at helping a reader of the ruleset help understand what is shown on the page (e.g. fully expose all the under the hood stuff that we don't really show in game help today).

We will keep these files updated as needed as we improve the code (e.g. add a new unit type class flag) like we do for other aspects of the game. If we update a shipped ruleset to take advantage of the new flag, we should process it with `sphinx_fc21_manuals` to update our RTD.

**Comment by lmoureaux** [comment]
Ok based on the discussion we had on Discord, I'd like to see pyciv mature a bit more before we integrate it into our workflows. Since this will take some time and I don't want to block this PR from adding good content, let me request the following:

* Remove all references to pyciv-generated documentation from this PR
* Integrate into the TOC and add some cross-links from other relevant pages (e.g. `effects.rst`) like normal fc21 docs would have

Then the overall strategy would be (some items can ofc happen in parallel):

* Get pyciv fully ready
* Figure out all the annoying bits about docs synchronization, CI, and so on (new concern: LT rulesets such as LTT/LTX living in other repos? [Intersphinx](https://docs.readthedocs.io/en/stable/guides/intersphinx.html) could help here). This happens in to-be-opened issues
* Include the generated docs
* Update instructions, CI, etc in the fc21 repo
* Add links to the generated ruleset docs from the main fc21 docs

Does that sound like a reasonable approach?

**Comment by jwrober** [comment]
Yes. I'll push another commit here to get the files added into the existing TOC with some other cross-links.

**Review by lmoureaux — COMMENTED** [review]
LGTM, few typos and a couple suggestions. The pages are very technical and may not be ideal for inclusion in a manual aimed at the lambda user.

Would it make sense to add links next to `TechFlag`, `BaseFlag`, `RoadFlag`, `ExtraFlag`, `UnitFlag`, `UnitClassFlag`, and `TerrainFlag` in `effects.rst`? Asking because I use that page a lot :D

File: docs/Manuals/Rulesets/Common/unit_type_flags.rst [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+by looking at the :file:`units.ruleset` file in any ruleset directory. The following flags are used in all
+of the rulesets shipped by the Freeciv21 developers: Alien, Civ1, Civ2, Civ2Civ3, Classic, Experimental,
+Granularity, Multiplayer, Royale, and Sandbox.
+
+:strong:`AirAttacker`
+  Very bad at attacking AEGIS.
+
+:strong:`Airbase`
+  This flag does not have a pre-set use. It's available for ruleset modders to add as a condition to an
+  :doc:`action enabler requirements vector </Modding/Rulesets/actions>`.
```
[the remark this answers is not in the document]
Comment (jwrober) [reply docs/Manuals/Rulesets/Common/unit_type_flags.rst:24]: where/what is `control.flags`?

**Comment by jwrober** [comment]
> Would it make sense to add links next to `TechFlag`, `BaseFlag`, `RoadFlag`, `ExtraFlag`, `UnitFlag`, `UnitClassFlag`, and `TerrainFlag` in `effects.rst`? Asking because I use that page a lot :D

I only see these associated to the table at the top. Is that what you are asking?

**Comment by lmoureaux** [comment]
> I only see these associated to the table at the top. Is that what you are asking?

Yes

File: docs/Manuals/Rulesets/Common/unit_type_flags.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (lmoureaux) [reply docs/Manuals/Rulesets/Common/unit_type_flags.rst:24]: https://github.com/longturn/freeciv21/blob/master/data/civ2civ3/units.ruleset#L28

**Comment by jwrober** [comment]
Ok I think I have all the requested changes pushed here.

**Review by lmoureaux — APPROVED** [review]

**Pull request merged**